GVRf/Framework/framework/src/main/jni/engine/renderer/batch.cpp

```diff
@@ -0,0 +1,141 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "batch.h"
+#include "objects/scene_object.h"
+#include "objects/components/camera.h"
+#include "objects/components/render_data.h"
+
+namespace gvr{
+Batch::Batch(): mesh_count(0), draw_count(0), vertex_count_(0), vao_dirty(true),
+		renderdata(nullptr){
+	vertices_.clear();
+    normals_.clear();
+    tex_coords_.clear();
+    indices_.clear();
+    matrices_.clear();
+    matrix_indices.clear();
+    renderdata = new RenderData();
+    mesh = new Mesh(); // should perhaps have a pool of meshes of a certain size
+}
+
+Batch::~Batch() {
+    vertices_.clear();
+    normals_.clear();
+    tex_coords_.clear();
+    indices_.clear();
+    matrices_.clear();
+    matrix_indices.clear();
```

Comment:
Place these calls in a new method (cleanup()?) and call from both locations. Not sure if cleanup is required in ctor.